### PR TITLE
consolidate constants of kubeadm to remove k/k dependencies

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind cleanup
/priority important-soon
/sig cluster-lifecycle

**What this PR does / why we need it**:  consolidate some constants under kubeadm, to allow us cutting dependencies on k/k

**Which issue(s) this PR fixes**: part of kubernetes/kubeadm#1638

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by dixudx — APPROVED** [review]
Looks good to me.
I will lgtm after the CIs become healthy.

**Review by neolit123 — COMMENTED** [review]
thanks
/approve

File: cmd/kubeadm/.import-restrictions [hunk 1]
```diff
@@ -121,7 +120,6 @@
 				"github.com/docker/go-connections/sockets",
 				"github.com/docker/go-connections/tlsconfig",
 				"github.com/docker/go-units",
-				"github.com/docker/libnetwork/ipvs",
```
Comment (yastij) [inline cmd/kubeadm/.import-restrictions]: this is still pulled through the following dependency:

`k8s.io/kubernetes/pkg/util/ipvs`-> `k8s.io/kubernetes/pkg/proxy/ipvs` -> `k8s.io/kubernetes/cmd/kubeadm/app/preflight`

TL;DR - we either need to copy `CanUseIPVSProxier` that is under `pkg/proxy/ipvs` or find a new home for it

File: cmd/kubeadm/app/constants/constants.go [hunk 2]
```diff
@@ -366,6 +366,34 @@ const (
 	// May be overridden by a flag at startup.
 	// Deprecated: use the secure KubeControllerManagerPort instead.
 	InsecureKubeControllerManagerPort = 10252
+
+	// IPVSProxyMode is match set up cluster with ipvs proxy model
+	// TODO (yastij): we should make this shared also with componentConfig
+	IPVSProxyMode = "ipvs"
```
Comment (rosti) [inline cmd/kubeadm/app/constants/constants.go]: Ideally these consts should be made part of the kube-proxy versioned API. I'll push for that in the next versions of it, but for now, this is a fine solution.

File: cmd/kubeadm/.import-restrictions (hunk 1 above)
Comment (rosti) [reply cmd/kubeadm/.import-restrictions]: I'll try to ping some kube-proxy folks about `k8s.io/kubernetes/pkg/proxy/ipvs`. This code needs to be shared and maintained by kube-proxy folks, so that it does not become outdated quickly. We should copy it as a last resort only.

Comment (yastij) [reply cmd/kubeadm/.import-restrictions]: @rosti - agreed

Comment (yastij) [reply cmd/kubeadm/.import-restrictions]: cc @kubernetes/sig-network-misc

Comment (rosti) [reply cmd/kubeadm/.import-restrictions]: Looking at the code, I think, that `GetKernelVersionAndIPVSMods` needs to be moved to `k8s.io/utils/net` and `CanUseIPVSProxier` to `k8s.io/kube-proxy/ipvs`.

**Comment by yastij** [comment]
/cc @andrewsykim 

for the ipvs bits

File: cmd/kubeadm/app/constants/constants.go [hunk 3]
```diff
@@ -366,6 +366,34 @@ const (
...
+
+	// ModeAlwaysAllow is the mode to set all requests as authorized
+	// TODO (yastij): we should make this shared also with componentConfig
+	ModeAlwaysAllow = "AlwaysAllow"
```
Comment (andrewsykim) [inline cmd/kubeadm/app/constants/constants.go]: Use [KubeletAuthorizationModeAlwaysAllow](https://github.com/kubernetes/kubelet/blob/master/config/v1beta1/types.go#L749)?

File: cmd/kubeadm/app/constants/constants.go [hunk 4]
```diff
@@ -366,6 +366,34 @@ const (
...
+	// TODO (yastij): we should make this shared also with componentConfig
+	ModeAlwaysDeny = "AlwaysDeny"
+
+	// ModeABAC is the mode to use Attribute Based Access Control to authorize
+	// TODO (yastij): we should make this shared also with componentConfig
+	ModeABAC = "ABAC"
+
+	// ModeWebhook is the mode to make an external webhook call to authorize
+	// TODO (yastij): we should make this shared also with componentConfig
+	ModeWebhook = "Webhook"
```
Comment (andrewsykim) [inline cmd/kubeadm/app/constants/constants.go]: Use [KubeletAuthorizationModeWebhook](https://github.com/kubernetes/kubelet/blob/master/config/v1beta1/types.go#L751)?

File: cmd/kubeadm/.import-restrictions (hunk 1 above)
Comment (andrewsykim) [reply cmd/kubeadm/.import-restrictions]: We're still calling CanUseIPVSProxier in this PR, wondering why the import restriction here passes now?

File: cmd/kubeadm/app/constants/constants.go (hunk 3 above)
Comment (andrewsykim) [reply cmd/kubeadm/app/constants/constants.go]: hmmm nvm, ignore that.. don't love the inconsistency

File: cmd/kubeadm/app/constants/constants.go (hunk 4 above)
Comment (andrewsykim) [reply cmd/kubeadm/app/constants/constants.go]: hmmm nvm, ignore that.. don't love the inconsistency

File: cmd/kubeadm/app/constants/constants.go (hunk 3 above)
Comment (andrewsykim) [reply cmd/kubeadm/app/constants/constants.go]: @liggitt @dims do you see a better home for `pkg/kubeapiserver/authorizer/modes`? Maybe `k8s.io/apiserver` or is that not generic enough?

File: cmd/kubeadm/.import-restrictions (hunk 1 above)
Comment (andrewsykim) [reply cmd/kubeadm/.import-restrictions]: nvm, it's failing :P

Comment (neolit123) [reply cmd/kubeadm/.import-restrictions]: responded that we should remove CanUseIPVSProxier on slack:
https://kubernetes.slack.com/archives/C13J86Z63/p1562973133411800

> k8s.io/kube-proxy/ipvs

seems like a good location, but i'm starting to think that we should call kube-proxy to tell us if the system is capable.
same goes for conversion and validations, @rosti .

**Comment by neolit123** [comment]
@yastij anything blocking us here?
we can still have the constants and work on the TODOs later, i guess.

**Comment by yastij** [comment]
@neolit123 - nothing blocking, I'll rebase and consolidate what's remaining

**Comment by yastij** [comment]
/close

As the consts were already moved + #81791 removed the ipvs const usage
